compiler/haskell-ide-core/src/Development/IDE/Compat.hs

```diff
@@ -0,0 +1,38 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE CPP #-}
+
+-- | Attempt at hiding the GHC version differences we can.
+module Development.IDE.Compat(
+    HieFile(..),
+    mkHieFile,
+    writeHieFile,
+    readHieFile
+    ) where
+
+#ifndef GHC_STABLE
+import HieBin
+import HieAst
+import HieTypes
+#else
+
+import GHC
+import GhcPlugins
+import NameCache
+import Avail
+import TcRnTypes
+
+
+mkHieFile :: ModSummary -> TcGblEnv -> RenamedSource -> Hsc HieFile
+mkHieFile = undefined
```

Comment:
That deserves a comment and a call to `error` or `fail` with a useful error message is probably also preferable to `undefined`.

Reply:
Oops, I forgot to fill in the dummy implementations after getting their types correct. Now done.